custom_components/plugwise/binary_sensor.py

```
@@ -130,16 +130,16 @@
-        for device_id in coordinator.new_devices:
-            device = coordinator.data.devices[device_id]
-            if not (binary_sensors := device.get(BINARY_SENSORS)):
+        for pw_entity_id in coordinator.new_pw_entities:
+            pw_entity = coordinator.data.entities[pw_entity_id]
+            if not (binary_sensors := pw_entity.get(BINARY_SENSORS)):
                 continue
             for description in PLUGWISE_BINARY_SENSORS:
                 if description.key not in binary_sensors:
                     continue
-                entities.append(PlugwiseBinarySensorEntity(coordinator, device_id, description))
+                ha_entities.append(PlugwiseBinarySensorEntity(coordinator, pw_entity, description))
```

Ruff (0.7.0)

Undefined name `ha_entities`

(F821)